_posts/2024-04-26-modules-101.markdown

```diff
@@ -0,0 +1,135 @@
+---
+layout: post
+title:  "Valkey modules 101"
+authors: 
+  - dmitrypol
+date: 2024-04-26 01:01:01 -0700
+categories: modules
+---
+
+## What are Valkey modules?  
+
+The idea of modules is to keep the core Valkey functionality lean and add extra features (such as new commands and data types) where needed.  Modules can be written in C or other languages that have C bindings.  In this article we will go through the process of building simple modules in C and Rust (using the Valkey Module Rust SDK).  This article expects the audience to be at least somewhat familiar with C, Rust and Valkey.  
```

Comment:
```suggestion
The idea of modules is to allow adding extra features (such as new commands and data types) to Valkey without making changes to the core code.  Modules are a special type of code distribution called a shared library, which can be loaded by other programs at runtime and executed.  Modules can be written in C or other languages that have C bindings.  In this article we will go through the process of building simple modules in C and Rust (using the Valkey Module Rust SDK).  This article expects the audience to be at least somewhat familiar with C, Rust and Valkey.  
```

@dmitrypol @stockholmux I would also like to suggest we enforce one sentence per line, it becomes much easier to review. 

The blog also assumes knowledge of git and cloning. I would suggest either adding more detailed structure for cloning Valkey so that it keeps the bar low.

Reply:
updated

Reply:
not sure what you mean by one sentence per line.  That will impact how the text is presented in the published blog.

Reply:
@dmitrypol Create a new line for each sentence (not in a code block). Markdown create paragraph breaks after two new lines, so it'll look fine in the resulting blog post.

Example:
```
This is my first sentence. 
This is my second sentence.

This is a new paragraph.
```

Will render like:
> This is my first sentence. This is my second sentence.
> 
> This is a new paragraph.

@madolson I _really_ need to setup a style guide and perhaps a action to check for these. I'll add it to the ever growing to-do list.

Reply:
I thought it does it too but when I make the change locally and run `bundle exec jekyll serve` it generates the break after one line. I will make the change and hopefully it will look fine